### PR TITLE
#49 in operator implementation

### PR DESCRIPTION
**Pull request by abynenkov-ib**

### CODE REVIEW

**Review by rchowinfoblox — APPROVED** [review]
LGTM

File: pkg/lexer/lexer.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ func (l *Lexer) NextToken() token.Token {
 			if isTypePattern(tok.Literal) {
 				tok.Type = token.TYPE_PATTERN
 			}
+			if isLongOperator(tok.Literal) {
```
Comment (wplu) [inline pkg/lexer/lexer.go:72]: should this be before isTypePattern?

**Review by wplu — APPROVED** [review]

File: pkg/lexer/lexer.go (hunk 1 above)
Comment (abynenkov-ib) [reply pkg/lexer/lexer.go:72]: Nope. isLongOperator is much more strict than isTypePattern

Comment (wplu) [reply pkg/lexer/lexer.go:72]: we need isLongOperator to take precedence over isTypePattern.  I misunderstood the code.  isLongOperator is taking precedence over isTypePattern when encountered.

Comment (wplu) [reply pkg/lexer/lexer.go:72]: The idiomatic code for precedence should be:
```
			if isLongOperator(tok.Literal) {
				tok.Type = token.LookupOperator(tok.Literal)
			} else if isTypePattern(tok.Literal) {
				tok.Type = token.TYPE_PATTERN
			}
			return tok
```

**Pull request merged**